signac/syncutil.py

```diff
@@ -48,13 +48,13 @@ def copytree(src, dst, copy_function=shutil.copy2, symlinks=False):
 
 
 class dircmp_deep(dircmp):
-
     def phase3(self):  # Find out differences between common files
         xx = filecmp.cmpfiles(self.left, self.right, self.common_files, shallow=False)
         self.same_files, self.diff_files, self.funny_files = xx
 
     methodmap = dict(dircmp.methodmap)
-    methodmap['samefiles'] = methodmap['diff_files'] = phase3
+    # The following line must be ignored: https://github.com/python/mypy/issues/708
+    methodmap['same_files'] = methodmap['diff_files'] = phase3  # type: ignore
```

Comment:
Explanation: This ignore is due to a [long-standing issue in mypy](https://github.com/python/mypy/issues/708). This is the ["third case" described by Guido van Rossum](https://github.com/python/mypy/issues/708#issuecomment-405796865) in that issue. 
```python
signac/syncutil.py:57: error: Incompatible types in assignment (expression has type "Callable[[dircmp_deep], Any]", target has type "Callable[[], None]")
```